cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -283,8 +289,28 @@ func Run(ctx context.Context, c *config.CompletedConfig) error {
 		}
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.CoordinatedLeaderElection) {
+
+		// TODO: This should be a shared informer factory
```

Comment:
it should :)

Reply:
don't you already have one from `controllerContext` here?

Reply:
@aojea @sttts I've been digging into this for a bit and want to double check a few things. 

`controllerContext` provides a SharedInformerFactory but this is created and started AFTER the leader is obtained. We need the informers ready beforehand to determine whether the kcm will be the leader.
ref: https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-controller-manager/app/controllermanager.go#L231
https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-controller-manager/app/controllermanager.go#L591

We could initialize a shared informer factory and reuse it in `controllerContext`, but this leads to the problem that we need to start the informerfactory BEFORE leader election. `controllerContext` initialization would then need to register additional informers and start the informerfactory again which seems like an antipattern. 

If the kcm is not elected as leader, we shouldn't initialize `controllerContext` or its set of informers. 

Based on these points, I think it would make sense to have stand alone informers for CoordinatedLeaderElection rather than attempting to reuse the sharedinformerfactory.